### PR TITLE
refac: Updated how grid area codes are anonymised + move data notebook

### PR DESCRIPTION
**Pull request by cenh-halfspace**
# Description
After talk with Mosaic added such that anonymised grid area codes cannot overlap with real ones. Also added a notebook for writing to a target table after having been moved.

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [x] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/exploratory-notebooks/data-anonymisation/anonymise_wholesale_output_data.py [hunk 1]
```diff
@@ -113,28 +123,31 @@
...
-    .withColumn(
-        anonymised_grid_area_code_column_name,
-        F.when(
-            F.col(grid_area_code_column_name).isNull(),
-            F.lit(None),
-        ).otherwise(F.col(anonymised_grid_area_code_column_name)),
-    )
-    .na.drop()
-).cache()
+df_all_grid_area_codes = df_source_grid_area_codes_table.union(df_source_energy_results_table.select(grid_area_code_column_name).distinct()).distinct()
```
Comment (JesperGomes) [inline source/databricks/exploratory-notebooks/data-anonymisation/anonymise_wholesale_output_data.py:131]: Where is df_source_energy_results_table defined?

File: source/databricks/exploratory-notebooks/data-anonymisation/anonymise_wholesale_output_data.py [hunk 2]
```diff
@@ -113,28 +123,31 @@
...
+
+anonymised_grid_area_codes = []
+list_unique_grid_area_codes = [row[grid_area_code_column_name] for row in df_all_grid_area_codes.collect()]
```
Comment (JesperGomes) [inline source/databricks/exploratory-notebooks/data-anonymisation/anonymise_wholesale_output_data.py:134]: Should the variable in line 134 be df_unique_grid_area_codes`?

File: source/databricks/exploratory-notebooks/data-anonymisation/anonymise_wholesale_output_data.py (hunk 1 above)
Comment (cenh-halfspace) [reply source/databricks/exploratory-notebooks/data-anonymisation/anonymise_wholesale_output_data.py:131]: On line 94

File: source/databricks/exploratory-notebooks/data-anonymisation/anonymise_wholesale_output_data.py (hunk 2 above)
Comment (cenh-halfspace) [reply source/databricks/exploratory-notebooks/data-anonymisation/anonymise_wholesale_output_data.py:134]: It is not a df, so it shouldn't be named df_, if you mean the variable in the list-comprehension, then it should take variable on line 131 as it contains all grid area codes.

Comment (JesperGomes) [reply source/databricks/exploratory-notebooks/data-anonymisation/anonymise_wholesale_output_data.py:134]: I meant line 131. It has a distinct, so isn't the difference between the two, that one is a list and the other is a df?

File: source/databricks/exploratory-notebooks/data-anonymisation/anonymise_wholesale_output_data.py (hunk 1 above)
Comment (cenh-halfspace) [reply source/databricks/exploratory-notebooks/data-anonymisation/anonymise_wholesale_output_data.py:131]: Sorry, read wrong line 86 is where df_source_energy_results_table is defined

**Review by JesperGomes — DISMISSED** [review]

**Pull request merged**